server/bleep/src/webserver/answer.rs

```diff
@@ -200,14 +209,16 @@ pub(super) async fn _handle(
         .clone()
         .into_owned();
 
+    exchanges.push(Exchange::new(query_id, query));
+
     let stream = async_stream::try_stream! {
         let mut action = Action::Query(query_target);
         let (exchange_tx, exchange_rx) = tokio::sync::mpsc::channel(10);
 
-        conversation.exchanges.push(Exchange::new(query));
         let mut agent = Agent {
             app,
-            conversation,
+            repo_ref: repo_ref.clone(),
+            exchanges: exchanges.clone(),
```

Comment:
Does this `clone()` not get expensive/large?

Reply:
Good spot, this was a relic